### PR TITLE
Destroying angel buff

### PR DESCRIPTION
**Pull request by wanderso**
## What Does This PR Do

Significant buff to both Destroying Angel mushrooms and to purge chemicals. 

Destroying Angel mushrooms, which mutate from amanita mushrooms, have the same primary poison (amanitin) but less of it than the 'base' amanita mushrooms (30% reagent gene in destroying angels, 35% in amanitas.) While they also have more potency than the base amanita, they have less yield as well, so even if we assume genes aren't being modified, destroying angels have less total poison (21u amanitin) than the fly amanita mushrooms (28u amanitin.) 

This changes Destroying Angels. Instead of being 30% amanitin, they're now 20% Nothing, 20% phalloidin (I looked up the name of the poison in actual destroying angel mushrooms. Kind of wish it wasn't named that). Phalloidin acts exactly the same as amanitin, but with two changes - one, it's invisible even to advanced health scanners, with the chemical not appearing on them. (A change has been made to permit other chemicals to be similarly invisible - worth considering whether or not things like changeling stings should be unable to be detected.) Two, it cannot be purged by charcoal, calomel, pentetic acid, or degreaser (if somehow someone was in a state where they could be affected by degreaser and phalloidin at the same time). Chemicals can also be immune to purges now. _Dialysis_, however, will still detect and remove phalloidin, so no poison is impossible to treat...

Because of this, I also added the 'immune to purge' trait to charcoal, calomel, and pentetic acid. Purging chems other than pentetic (or just stuffing someone full of 80u charcoal if the sleeper is free) are currently not used because they'd purge each other - this allows charcoal, calomel, and pentetic acid to stack, using all of them at once if someone is staggering into medbay with 180u of spider venom inside them. 

Also, Destroying Angel mushrooms can be used in place of Ambrosia Gaia to reach Biotech 7... if you're okay with leaving such an extremely stealthy poison lying on the desk of science, that is...

## Why It's Good For The Game
Destroying Angel mushrooms currently have no function, being less toxic than the amanitin they are mutated from, an obvious sign that someone in botany is up to no good, and generally being strictly worse than amanitin in every way, shape, and form. This creates a new, stealthier poison. Also, to make other poisons easier to fight, this buffs purge chemicals by keeping them from purging each other.

## Images of changes
image 
_A new reagent gene..._
image 
_Not immune to dialysis!_
image
_Nothing bad will happen if we let them grow this. It's for science._

## Testing

Extensively tested it on my local machine. I wanted to make phalloidin break down into [current_cycle]u of amanitin, so that the amanitin could be detected and purged, but that caused crashes for some reason, so that's listed as a 'TODO' in the future. Gave someone a bunch of different purge chemicals at once and verified that everything that wasn't phalloidin or another purge chemical was purged, and that they were all visible at the same time.

## Changelog
:cl:
add: Added phalloidin, a new toxin produced by the destroying angel mushroom almost impossible to treat and detect
tweak: Changed it so that purge chemicals no longer purged each other 
wip: Would like to make phalloidin break down into amanitin, possibly make changeling stings similarly hard to detect
/:cl:

### CODE REVIEW

**Comment by Fordoxia** [comment]
The real buff here is the buff to anti-toxin medication. And I'm all for it tbh.

**Comment by matttheficus** [comment]
Botany does not need another deathchem - especially one that is immune to treatment. The stealth aspect is good enough without making it an AoE deathbutton. If you REALLY want it to be “harder to treat” make it ignore charcoal only, perhaps. Pentetic should still clear it.

If you want a variant to be immune, it needs to be an Uplink toxin. Make it Botany-specific if ya feel.

**Comment by SchrodingersWolf** [comment]
Really not a big fan of phalloidin in its current iteration - A completely undetectable poison that has the potential to instakill with no warning, which also cannot be purged without a second person's involvement, and is also readily available in Botany?

There is a point about poisons not being stealthy enough, but considering that almost every instakill method has a way to be detected, however short the window is. (The only way I know to kill someone without them knowing at all is a doubletap with a sleepypen that contains explosive chems) The fact that dialysis can remove it is immaterial because you don't even know it is there to begin with.

As for purge chems stacking, not very happy about calomel+charcoal, being roundstart chems, becoming almost as good as pentetic - I also kind of liked the fact that you have to choose between the different purge chems instead of just injecting all of them in. (Pentetic does dominate that choice, but I feel like nerfing/changing pentetic is the right call over removing that choice altogether) 

But hey, at least I don't have to play 'Dodge that Medibot' with my patient when I have given them pentetic.

**Comment by CinnamonSnowball** [comment]
> Because of this, I also added the 'immune to purge' trait to charcoal, calomel, and pentetic acid. Purging chems other than pentetic (or just stuffing someone full of 80u charcoal if the sleeper is free) are currently not used because they'd purge each other - this allows charcoal, calomel, and pentetic acid to stack

This change seems fairly unrelated to the whole botany mushroom deal, it should be in a separate PR

**Comment by wanderso** [comment]
> Botany does not need another deathchem - especially one that is immune to treatment. The stealth aspect is good enough without making it an AoE deathbutton. If you REALLY want it to be “harder to treat” make it ignore charcoal only, perhaps. Pentetic should still clear it.
> 
> If you want a variant to be immune, it needs to be an Uplink toxin. Make it Botany-specific if ya feel.

Originally, as mentioned, I wanted phalloidin to deal no damage on its own, but to instead convert into amanitin upon leaving the body (so, yes, you had to wait for the phalloidin to leave the system, and _then_ wait for the amanitin to leave the system before it dealt any direct damage at all.) The amanitin could then be detected and purged normally. Unfortunately, doing that resulted in a crash.

Would the original plan help your objections any?

**Comment by wanderso** [comment]
> > Because of this, I also added the 'immune to purge' trait to charcoal, calomel, and pentetic acid. Purging chems other than pentetic (or just stuffing someone full of 80u charcoal if the sleeper is free) are currently not used because they'd purge each other - this allows charcoal, calomel, and pentetic acid to stack
> 
> This change seems fairly unrelated to the whole botany mushroom deal, it should be in a separate PR to keep things atomic

You're correct. I'll make it a separate PR later today.

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 1]
```diff
@@ -913,6 +913,20 @@
...
+	id = "phalloidin"
+	description = "A toxin produced by Destroying Angel mushrooms. Deadly, stealthy, and impossible to purge."
+	reagent_state = LIQUID
+	color = "#E7E7E7"
+	taste_mult = 0
+	unable_to_purge = TRUE
+	invisible_to_scanners = TRUE
+
+/datum/reagent/phalloidin/on_mob_delete(mob/living/M)
+	M.adjustToxLoss(current_cycle*rand(2,4))
```
Comment (Sirryan2002) [inline code/modules/reagents/chemistry/reagents/toxins.dm:927]:
```suggestion
	M.adjustToxLoss(current_cycle * rand(2,4))
```

**Comment by matttheficus** [comment]
> > Botany does not need another deathchem - especially one that is immune to treatment. The stealth aspect is good enough without making it an AoE deathbutton. If you REALLY want it to be “harder to treat” make it ignore charcoal only, perhaps. Pentetic should still clear it.
> > If you want a variant to be immune, it needs to be an Uplink toxin. Make it Botany-specific if ya feel.
> 
> Originally, as mentioned, I wanted phalloidin to deal no damage on its own, but to instead convert into amanitin upon leaving the body (so, yes, you had to wait for the phalloidin to leave the system, and _then_ wait for the amanitin to leave the system before it dealt any direct damage at all.) The amanitin could then be detected and purged normally. Unfortunately, doing that resulted in a crash.
> 
> Would the original plan help your objections any?

I’m unsure if this really does anything then - why not just have it given more amanitin? You’re likely going to go to Medbay just in case either way and people will learn that “oh it’s probably just super-invisible-amanitin” causing them to sit AFK in Medbay till it procs. That doesn’t sound enjoyable IMO.

**Comment by Komrad822** [comment]
> Botany does not need another deathchem - especially one that is immune to treatment. The stealth aspect is good enough without making it an AoE deathbutton. If you REALLY want it to be “harder to treat” make it ignore charcoal only, perhaps. Pentetic should still clear it.
> 
> If you want a variant to be immune, it needs to be an Uplink toxin. Make it Botany-specific if ya feel.

My thoughts exactly. Currently the chem is too bloated to be in a plant that can be mass produced. People should be able to counter things that are easy to produce in mass amounts(definition of botaniy's strength)

File: code/modules/hydroponics/grown/mushrooms.dm [hunk 2]
```diff
@@ -77,17 +77,17 @@
...
 
 /obj/item/food/snacks/grown/mushroom/angel
 	seed = /obj/item/seeds/angel
 	name = "destroying angel"
-	desc = "<I>Amanita Virosa</I>: Deadly poisonous basidiomycete fungus filled with alpha amanitin."
+	desc = "<I>Amanita Virosa</I>: Deadly poisonous basidiomycete fungus filled with phalloidin."
 	icon_state = "angel"
 	filling_color = "#C0C0C0"
 	tastes = list("destroying angel" = 1)
+	origin_tech = "biotech=6" // as useful as ambrosia gaia, and quicker to get - but you have to leave traitor tools lying around...
```
Comment (Spaghetti-bit) [inline code/modules/hydroponics/grown/mushrooms.dm:90]: Not sure I understand the comment, traitor tools? I don't think a single destroying angel will useful enough for a traitor. Perhaps single syringe or pill. Or poison. But they have access to other tools easily enough.

File: code/modules/reagents/chemistry/reagents/medicine.dm [hunk 3]
```diff
@@ -401,13 +401,15 @@
 	color = "#000000"
 	taste_description = "dust"
 	goal_difficulty = REAGENT_GOAL_EASY
+	unable_to_purge = TRUE
```
Comment (Spaghetti-bit) [inline code/modules/reagents/chemistry/reagents/medicine.dm:404]: perhaps rename to `purgeable` for readability.



Example:

`R.purgeable` vs. `R.unable_to_purge`

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 1 above)
Comment (Spaghetti-bit) [inline code/modules/reagents/chemistry/reagents/toxins.dm:927]: Balance wise, perhaps tune this down to `rand(1,3)` and allow it to do minor brain damage.

File: code/modules/reagents/chemistry/reagents_datum.dm [hunk 4]
```diff
@@ -37,6 +37,8 @@
 	var/taste_description = "metaphorical salt"
 	/// how quickly the addiction threshold var decays
 	var/addiction_decay_rate = 0.01
+	var/invisible_to_scanners = FALSE // If TRUE, this cannot be detected inside a body with the advanced health scanner's reagent scan
+	var/unable_to_purge = FALSE // If TRUE, this cannot be purged with mercury, calomel, or charcoal
```
Comment (Spaghetti-bit) [inline code/modules/reagents/chemistry/reagents_datum.dm:41]: Like suggestion above, rename to `purgeable`. Also don't forget to auto doc the variables you made with `/// stuff`

File: code/modules/reagents/chemistry/reagents_datum.dm [hunk 5]
```diff
@@ -37,6 +37,8 @@
 	var/taste_description = "metaphorical salt"
 	/// how quickly the addiction threshold var decays
 	var/addiction_decay_rate = 0.01
+	var/invisible_to_scanners = FALSE // If TRUE, this cannot be detected inside a body with the advanced health scanner's reagent scan
```
Comment (Spaghetti-bit) [inline code/modules/reagents/chemistry/reagents_datum.dm:40]: Invert to `var/visible_to_scanners = TRUE` then set the corresponding chemicals tofalse where need be.

**Review by Spaghetti-bit — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/grown/mushrooms.dm (hunk 2 above)
Comment (Spaghetti-bit) [reply code/modules/hydroponics/grown/mushrooms.dm:90]: With a small balance tweak perhaps? Botany doesn't need more killing tools.

**Pull request closed without merging**